code/game/objects/items/stacks/stack.dm

```diff
@@ -10,156 +10,171 @@
  */
 /obj/item/stack
 	origin_tech = "materials=1"
-	var/list/datum/stack_recipe/recipes
+	var/list/recipes = list() // /datum/stack_recipe
 	var/singular_name
 	var/amount = 1
 	var/max_amount //also see stack recipes initialisation, param "max_res_amount" must be equal to this max_amount
 
-/obj/item/stack/New(var/loc, var/amount=null)
+/obj/item/stack/New(var/loc, var/amt = null)
 	..()
-	if(amount!=null)	//Allow for stacks with the amount=0
-		src.amount=amount
-	return
+	if(amt != null)	//Allow for stacks with the amount=0
+		amount = amt
 
 /obj/item/stack/Destroy()
-	if(usr && usr.machine==src)
+	if(usr && usr.machine == src)
 		usr << browse(null, "window=stack")
 	return ..()
 
 /obj/item/stack/examine(mob/user)
 	if(..(user, 1))
-		to_chat(user, "There are [src.amount] [src.singular_name]\s in the stack.")
+		to_chat(user, "There are [amount] [singular_name]\s in the stack.")
 
-
-/obj/item/stack/attack_self(mob/user as mob)
+/obj/item/stack/attack_self(mob/user)
 	list_recipes(user)
 
-/obj/item/stack/proc/list_recipes(mob/user as mob, recipes_sublist)
+/obj/item/stack/proc/list_recipes(mob/user, recipes_sublist)
 	if(!recipes)
 		return
-	if(!src || amount<=0)
+
+	if(amount <= 0)
 		user << browse(null, "window=stack")
+		return
+
 	user.set_machine(src) //for correct work of onclose
+
 	var/list/recipe_list = recipes
 	if(recipes_sublist && recipe_list[recipes_sublist] && istype(recipe_list[recipes_sublist], /datum/stack_recipe_list))
 		var/datum/stack_recipe_list/srl = recipe_list[recipes_sublist]
 		recipe_list = srl.recipes
+
 	var/t1 = "Amount Left: [amount]<br>"
-	for(var/i=1;i<=recipe_list.len,i++)
+	for(var/i = 1, i <= recipe_list.len, i++)
```

Comment:
@Fox-McCloud I have literally no reason to take any advice from you on this, but fine, whatever.